content/post/pga_history.md

```diff
@@ -0,0 +1,117 @@
+a Gerrit review.
+* [Git Bomb](https://github.com/Katee/git-bomb) - an "uncloneable" repository. By the way, we cloned it.
+* Some repositories have packfiles in the [old deprecated format]. go-git handles them.(https://github.com/git/git/blob/master/Documentation/technical/pack-format.txt#L39).
+* Load balancing is difficult. 90% of the repositories are compact and fly fast, however, the rest 10% occupy 50% of the dataset size and are damn slow to process.
+* Analyzing big repositories requires much memory, as much as 128 GB. Thus our pipeline kept exploding again and again and we restarted it with a firm hand.
+
+We had to take into account the removed repositories. Over 3,000 were deleted within few months.
+We discovered a few repositories with porno while inspecting the outliers by repository size (no links here, sorry, join our community Slack and direct message).
+
+Overall, our Data Retrieval team did a fantastic job. It was great to see how the numerous open source libraries they developed played together to deliver the dataset.
+
+#### Paper
+
+We used [Overleaf](https://www.overleaf.com/) to write the paper. When we reached the internal review
+phase, we pushed it to a private GitHub repository and the rest of the edits happenned in pull requests.
+Thus we avoided the "cold start" problem where the first pull request is huge and nightmare to read and collaborate. If we did not have to use TeX, we would use Google Docs instead.
+
+It was hard to squeeze the content into 4 pages. The layout constantly broke, we had to "fix" it with
+`\vspace`. Figures tended to appear on a different page from the related text so we forced the placement
+and got the "jumping" problem instead. The bibliography was very verbose and we had to "optimize" the records
+to make them shorter and consequently fit into one column.
+
+#### Poster
+
+We included the QR code which links to bit.ly into the poster to track the scans.
+This is how it looked like on the conference:
+
+{{% caption src="/post/pga_history/poster.jpg" %}}
+Conference attendees looking at the PGA poster on MSR'18/ICSE'18.
+{{% /caption %}}
+
+#### Licensing
+
+The [GDPR law](https://www.eugdpr.org/) in EU enforces strict rules of dealing with the private information.
+Git repositories contain commits and the commit metadata contains people's names, email addresses and
+timestamps which allow identification. We've discovered that while the code itself may be OSS,
+its license does not extend to the development history. In other words,
+Git repositories distribution looks like a GDPR infringement. However, there is
+an [exclusion in GDPR](https://news.ycombinator.com/item?id=16509688) which really saves us:
+Git metadata is an inherent part of the dataset because anonymization changes the Git commit
+hashes. Besides, PGA does not contain private information in an easily accessible format because
+you still need to read the Git packfiles to extract it.
+
+GHTorrent is licensed under CC-BY-SA.
+
```

Comment:
I would remove the Paper, Poster, Cut scenes and Licensing sections above to shorten this blog post and focus on the retrieval problems & origin of PGA.